pkg/statsd/handler_http_forwarder_v2.go

```diff
@@ -212,15 +217,18 @@ func (hfh *HttpForwarderHandlerV2) Run(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		case metricMaps := <-hfh.consolidatedMetrics:
-			if !hfh.acquireSem(ctx) {
-				return
+			mergedMetricMap := mergeMaps(metricMaps)
+			mms := mergedMetricMap.SplitByTags(hfh.dynHeaders)
+			for dynHeaderTags, mm := range mms {
+				if !hfh.acquireSem(ctx) {
```

Comment:
I may not understand what's going on here fully but something about acquiring the lock during a loop and releasing it in an async go func below seems wrong to me. Seems like it would do the first one and then if the go func didn't execute fast enough just return and skip the rest of the loop iterations?

Reply:
It's a semaphore, to stop a flood of goroutines. In practice, for a forwarder, there's not a high volume of requests anyway. It's more valuable at the aggregator tier, where you might have 500 requests going to a vendor, and want to limit that.

As much as possible, there's explicit back pressure in the system (ultimately ending at not reading from the network stack and dropping udp statsd packets).

Reply:
Ah I see, so acquireSem will block and not return false. The not is an error condition, gotcha.

Reply:
Correct - the only time it returns is when `ctx` is canceled, which only occurs on shutdown.

Reply:
We'll probably need to review the value for `max-requests` when deploying forwarder as a daemonset.